src/Process/Runner/ParallelProcessRunner.php

```diff
@@ -213,22 +215,27 @@ private function startProcess(MutantProcessContainer $mutantProcessContainer, in
      *  - from the input stream of processes containers (original mutant processes)
      *  - from the "next" killer processes, created if a PHPUnit process doesn't kill a Mutant
      *
-     * @param MutantProcessContainer[] $bucket
-     * @param Generator<MutantProcessContainer> $input
+     * @param array<MutantProcessContainer> $bucket
+     * @param Iterator<MutantProcessContainer> $input
      */
-    private function fillBucketOnce(array &$bucket, Generator $input, int $threadCount): int
+    private function fillBucketOnce(array &$bucket, Iterator $input, int $threadCount): int
     {
         if (count($bucket) >= $threadCount || !$input->valid()) {
             if ($this->nextMutantProcessKillerContainer !== []) {
-                $bucket[] = array_shift($this->nextMutantProcessKillerContainer);
+                $nextProcess = array_shift($this->nextMutantProcessKillerContainer);
+                Assert::notNull($nextProcess);
+                $bucket[] = $nextProcess;
             }
 
             return 0;
         }
 
         $start = microtime(true);
 
-        $bucket[] = $input->current();
+        $current = $input->current();
+        Assert::notNull($current);
```

Comment:
`$input->current()` will return `null` if the generator is not yet `valid()`  - exactly what happens if we remove the `return 0` above; Really we should switch this whole thing to SplQueue or some other more appropriate tool.